### PR TITLE
Move TransactionPool from 'common' to 'node'

### PR DESCRIPTION
**Pull request by Geod24**
It is not a common utility, and very much tied to the node.
As a rule of thumb, common should not import from node / consensus.

### CODE REVIEW

**Review by AndrejMitrovic — APPROVED** [review]

**Review by linked0 — APPROVED** [review]

**Review by kchulj — APPROVED** [review]
LGTM

**Comment by Geod24** [comment]
Oh, this would conflict big time with #1464 so let's wait until that one is merged first.

**Comment by Geod24** [comment]
Rebased

**Pull request merged**
